File: NetLibraryGenerator/NetLibraryGenerator/Model/Results/GenerationResults.cs
using Newtonsoft.Json;

namespace NetLibraryGenerator.Model.Results;

public class GenerationResults
{
    private GenerationResults() {}
    public static readonly GenerationResults Instance = new GenerationResults();
    
    [JsonProperty("changedMethods")]
    public List<ChangedMethod> ChangedMethods { get; } = new();
    
    [JsonProperty("addedMethods")]
    public List<AddedMethod> AddedMethods { get; } = new();
    
    [JsonProperty("addedCategories")]
    public List<string> AddedCategories { get; } = new();

    [JsonProperty("modelEntitiesCount")]
    public int ModelEntitiesCount { get; set; }
    
    [JsonProperty("modelEventsCount")]
    public int ModelEventsCount { get; set; }
    
    [JsonProperty("modelResponsesCount")]
    public int ModelResponsesCount { get; set; }
    
    [JsonProperty("modelParametersCount")]
    public int ModelParametersCount { get; set;}
    
    [JsonProperty("modelEnumsCount")]
    public int ModelEnumsCount { get; set; }
}

File: DartLibraryGenerator/model/results/generation_results.dart
import 'added_method.dart';
import 'changed_method.dart';

class GenerationResults {
    GenerationResults._();

    static int modelEntitiesCount = 0;
    static int modelEventsCount = 0;
    static int modelResponsesCount = 0;
    static int modelParametersCount = 0;
    static int modelEnumsCount = 0;
    
    static final List<ChangedMethod> changedMethods = [];
    static final List<AddedMethod> addedMethods = [];
    static final List<String> addedCategories = [];
    
    static Map<String, dynamic> toJson() => {
        "changedMethods": List<dynamic>.from(changedMethods.map((x) => x.toJson())),
        "addedMethods": List<dynamic>.from(addedMethods.map((x) => x.toJson())),
        "addedCategories": List<dynamic>.from(addedCategories.map((x) => x)),
        "modelEntitiesCount": modelEntitiesCount,
        "modelEventsCount": modelEventsCount,
        "modelResponsesCount": modelResponsesCount,
        "modelParametersCount": modelParametersCount,
        "modelEnumsCount": modelEnumsCount,
    };
}